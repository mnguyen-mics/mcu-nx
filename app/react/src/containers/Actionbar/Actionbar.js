import React from 'react';
import PropTypes from 'prop-types';
import { Row } from 'antd';
import BreadcrumbBar from './BreadcrumbBar';

function Actionbar(props) {
  return (
    <Row
      type="flex"
      align="middle"
      justify="space-between"
      className={'mcs-actionbar-edit'}
    >
      <BreadcrumbBar {...props} className={'mcs-breadcrumb-edit'} />
      <div className="left-part-margin">
        {props.children}
      </div>
    </Row>
  );
}

Actionbar.defaultProps = {
  edition: false,
};

Actionbar.propTypes = {
  edition: PropTypes.bool,
};

export default Actionbar;
