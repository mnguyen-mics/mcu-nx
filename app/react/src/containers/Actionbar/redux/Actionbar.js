
const ACTION_BAR = {
  BREADCRUMB: {
    PUSH: 'ACTION_BAR_BREADCRUMB_PUSH',
    POP: 'ACTION_BAR_BREADCRUMB_POP',
    SET: 'ACTION_BAR_BREADCRUMB_SET',
  },
  SET_SECONDARY: 'ACTION_BAR_SET_SECONDARY'
};

const setSecondaryActionBar = visible => dispatch =>
    dispatch({
      type: ACTION_BAR.SET_SECONDARY,
      secondaryBar: visible
    });

const pushBreadcrumb = breadcrumbValue => dispatch =>
  dispatch({
    type: ACTION_BAR.BREADCRUMB.PUSH,
    breadcrumb: { breadcrumbName: breadcrumbValue }
  });

const popBreadcrumb = () => dispatch =>
  dispatch({
    type: ACTION_BAR.BREADCRUMB.POP
  });

const setBreadcrumb = (from, path) => dispatch =>
  dispatch({
    type: ACTION_BAR.BREADCRUMB.SET,
    from,
    path
  });


const defaultActionbarState = {
  path: [],
  secondary: null
};


const actionbarState = (state = defaultActionbarState, action) => {
  switch (action.type) {
    case ACTION_BAR.BREADCRUMB.PUSH: {
      const newPath = [...state.path];
      newPath.push(action.breadcrumb);
      return {
        ...state,
        path: newPath
      };
    }

    case ACTION_BAR.BREADCRUMB.POP: {
      const newPath = [...state.path];
      newPath.pop();
      return {
        ...state,
        path: newPath
      };
    }

    case ACTION_BAR.BREADCRUMB.SET: {

      const newPath = action.from === 0 ?
        action.path : state.path.slice(0, action.from).concat(action.path);

      return {
        ...state,
        path: newPath
      };
    }

    case ACTION_BAR.SET_SECONDARY: {
      return {
        ...state,
        secondary: action.secondaryBar
      };
    }

    default:
      return state;
  }
};

export {

  actionbarState,

  setSecondaryActionBar,

  pushBreadcrumb,
  popBreadcrumb,
  setBreadcrumb
};
