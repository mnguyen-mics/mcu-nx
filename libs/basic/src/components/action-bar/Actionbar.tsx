import React from 'react';
import { Row } from 'antd';
import Breadcrumb, { BreadcrumbProps } from 'antd/lib/breadcrumb';
import McsIcon from '../mcs-icon';

const BreadcrumbItem = Breadcrumb.Item;

export interface ActionbarProps extends BreadcrumbProps {
  className?: string;
  edition?: boolean;
  pathItems: React.ReactNode[];
  backgroundColor?: string;
  inverted?: boolean;
}

type Props = ActionbarProps;

class Actionbar extends React.Component<Props> {
  render() {
    const { edition, pathItems, backgroundColor, inverted, className, children, ...rest } =
      this.props;

    return (
      <Row
        align='middle'
        justify='space-between'
        className={`${edition ? 'mcs-actionbar-edit' : 'mcs-actionbar'} ${
          inverted ? 'inverted' : ''
        } ${className ? className : ''}`}
        style={backgroundColor ? { backgroundColor } : {}}
      >
        <Breadcrumb
          className={'mcs-breadcrumb'}
          separator={<McsIcon type='chevron-right' />}
          {...rest}
        >
          {pathItems.map((item, index) => (
            <BreadcrumbItem key={index}>{item}</BreadcrumbItem>
          ))}
        </Breadcrumb>
        <div className='left-part-margin'>{children}</div>
      </Row>
    );
  }
}

export default Actionbar;
