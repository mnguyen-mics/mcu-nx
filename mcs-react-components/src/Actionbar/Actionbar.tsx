import * as React from 'react';
import { Row } from 'antd';
import BreadcrumbBar, {BreadcrumbBarProps} from './BreadcrumbBar';
import {FormattedMessage} from 'react-intl';

export interface ActionbarProps extends BreadcrumbBarProps {
  edition?: boolean,
}

const Actionbar : React.SFC<ActionbarProps> = props  => {

  const { edition } = props;

  return (
    <Row
      type="flex"
      align="middle"
      justify="space-between"
      className={edition ? 'mcs-actionbar-edit' : 'mcs-actionbar'}
    >
      <BreadcrumbBar
        {...props}
        className={edition ? 'mcs-breadcrumb-edit' : 'mcs-breadcrumb'}
      />
      <div className="left-part-margin">
        {props.children}
      </div>
    </Row>
  );
};

export const SimpleActionBar = (title: FormattedMessage.MessageDescriptor) : React.SFC<any> => () => {
  return Actionbar({paths: [{name: title}]})
};


export default Actionbar;
