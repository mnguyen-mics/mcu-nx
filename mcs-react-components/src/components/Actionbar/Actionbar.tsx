import * as React from 'react';
import { Row } from 'antd';
import * as cuid from 'cuid';
import { compose } from 'recompose';
import { Link } from 'react-router-dom';
import { FormattedMessage, injectIntl, InjectedIntlProps } from 'react-intl';
import Breadcrumb, { BreadcrumbProps } from 'antd/lib/breadcrumb';
import McsIcon from '../Icon';

const BreadcrumbItem = Breadcrumb.Item;

export interface Path {
  name: FormattedMessage.MessageDescriptor | string;
  path?: string;
}

export interface ActionbarProps extends BreadcrumbProps {
  edition?: boolean;
  paths: Path[];
  backgroundColor?: string;
  inverted?: boolean;
}

type Props = ActionbarProps & InjectedIntlProps;

class Actionbar extends React.Component<Props> {
  buildItem = (elt: Path) => {
    const name =
      typeof elt.name === 'string'
        ? elt.name
        : this.props.intl.formatMessage(elt.name);
    const formatedElt = name
      ? name.substr(0, 27) !== name
        ? `${name.substr(0, 27)}\u2026`
        : name
      : null;
    const item = elt.path ? (
      <Link to={elt.path}>{formatedElt}</Link>
    ) : (
      formatedElt
    );

    return <BreadcrumbItem key={cuid()}>{item}</BreadcrumbItem>;
  };

  render() {
    const {
      edition,
      paths,
      intl,
      backgroundColor,
      inverted,
      children,
      ...rest
    } = this.props;
    return (
      <Row
        type="flex"
        align="middle"
        justify="space-between"
        className={`${edition ? 'mcs-actionbar-edit' : 'mcs-actionbar'} ${
          inverted ? 'inverted' : ''
        }`}
        style={backgroundColor ? { backgroundColor } : {}}
      >
        <Breadcrumb
          className={edition ? 'mcs-breadcrumb-edit' : 'mcs-breadcrumb'}
          separator={<McsIcon type="chevron-right" />}
          {...rest}
        >
          {paths.map(this.buildItem)}
        </Breadcrumb>

        <div className="left-part-margin">{children}</div>
      </Row>
    );
  }
}

export default compose<Props, ActionbarProps>(injectIntl)(Actionbar);
