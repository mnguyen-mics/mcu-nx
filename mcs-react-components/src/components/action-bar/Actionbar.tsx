import * as React from 'react';
import { Row } from 'antd';
import Breadcrumb, { BreadcrumbProps } from 'antd/lib/breadcrumb';
import McsIcon from '../mcs-icon';

const BreadcrumbItem = Breadcrumb.Item;

export interface ActionbarProps extends BreadcrumbProps {
  edition?: boolean;
  pathItems: React.ReactNode[];
  backgroundColor?: string;
  inverted?: boolean;
}

type Props = ActionbarProps;

class Actionbar extends React.Component<Props> {

  render() {
    const {
      edition,
      pathItems,
      backgroundColor,
      inverted,
      children,
      ...rest
    } = this.props;

    return (
      <Row
        align="middle"
        justify="space-between"
        className={`${edition ? 'mcs-actionbar-edit' : 'mcs-actionbar'} ${inverted ? 'inverted' : ''
          }`}
        style={backgroundColor ? { backgroundColor } : {}}
      >
        <Breadcrumb
          className={'mcs-breadcrumb'}
          separator={<McsIcon type="chevron-right" />}
          {...rest}
        >
          {pathItems.map((item, index) => <BreadcrumbItem key={index}>{item}</BreadcrumbItem>)}
        </Breadcrumb>
        <div className="left-part-margin">{children}</div>
      </Row>
    );
  }
}

export default Actionbar;
