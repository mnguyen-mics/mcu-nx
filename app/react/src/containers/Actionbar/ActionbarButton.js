
import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import classNames from 'classnames';
import { ActionbarActions } from './redux';

class ActionbarButton extends Component {

  constructor(props) {
    super(props);
    this.toggleSecondaryBar = this.toggleSecondaryBar.bind(this);
  }

  toggleSecondaryBar() {
    const {
      secondaryBar,
      setSecondaryActionBar,
      currentSecondaryBar,
    } = this.props;

    if (currentSecondaryBar !== null && currentSecondaryBar === secondaryBar) {
      setSecondaryActionBar(null);
    } else {
      setSecondaryActionBar(secondaryBar);
    }
  }

  render() {

    const {
      text,
      imageUrl,
      containerClassNames,
      buttonClassNames,
      secondaryBar,
      currentSecondaryBar,
      setSecondaryActionBar,
      onClick,
      ...other
    } = this.props;

    const content = text ? (imageUrl ? <img src={imageUrl} alt={text} /> : text) : this.props.children;

    const selected = currentSecondaryBar !== null && currentSecondaryBar === secondaryBar;

    const selectionClasses = ['selectionArrow'];

    if (!selected) { selectionClasses.push('hidden'); }


    const clickAction = evt => {
      this.toggleSecondaryBar();
      if (onClick) { onClick(evt); }
    };

    return (
      <div className={classNames(containerClassNames)}>
        <button className={classNames(buttonClassNames)} onClick={clickAction} {...other}>
          {content}
        </button>
        <svg className={classNames(selectionClasses)} width="40" height="10">
          <polyline
            points="0 7 10 0 20 7" stroke="#d3dbe1" strokeWidth="1"
            strokeLinecap="butt" fill="none"
          />
        </svg>
      </div>
    );
  }
}

ActionbarButton.propTypes = {
  text: PropTypes.string,
  imageUrl: PropTypes.string,
  secondaryBar: PropTypes.string,
  containerClassNames: PropTypes.arrayOf(PropTypes.string),
  buttonClassNames: PropTypes.arrayOf(PropTypes.string),
  setSecondaryActionBar: PropTypes.func, // eslint-disable-line react/require-default-props
  currentSecondaryBar: PropTypes.string, // eslint-disable-line react/require-default-props
  onClick: PropTypes.func // eslint-disable-line react/require-default-props
};

ActionbarButton.defaultProps = {
  buttonClassNames: ['mcs-actionbar-button'],
  containerClassNames: ['mcs-actionbar-button-wrapper'],
  secondaryBar: null,
  imageUrl: null,
  text: null
};

const mapStateToProps = state => ({
  currentSecondaryBar: state.actionbarState.secondary
});
const mapDispatchToProps = {
  setSecondaryActionBar: ActionbarActions.setSecondaryActionBar
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ActionbarButton);
