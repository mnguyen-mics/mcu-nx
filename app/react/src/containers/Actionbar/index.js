import Actionbar from './Actionbar';
import Editbar from './Editbar';
import ActionbarButton from './ActionbarButton';
import SecondaryActionbar from './SecondaryActionbar';

export {
  Actionbar,
  Editbar,
  ActionbarButton,
  SecondaryActionbar
};
