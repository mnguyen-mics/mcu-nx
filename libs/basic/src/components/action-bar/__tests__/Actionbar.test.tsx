import 'jest';
import React from 'react';
import TestRenderer from 'react-test-renderer';
import Actionbar, { ActionbarProps } from '../Actionbar';

it('renders the actionBar', () => {
  const props: ActionbarProps = {
    pathItems: [
      <a key='1' href='https://www.google.fr'>
        Campaigns
      </a>,
      <a key='2' href='https://www.github.fr'>
        Display
      </a>,
    ],
  };

  const component = TestRenderer.create(<Actionbar {...props} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
