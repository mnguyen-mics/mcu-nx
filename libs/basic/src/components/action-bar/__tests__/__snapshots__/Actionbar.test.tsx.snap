// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the actionBar 1`] = `
<div
  className="ant-row ant-row-space-between ant-row-middle mcs-actionbar  "
  style={Object {}}
>
  <nav
    className="ant-breadcrumb mcs-breadcrumb"
  >
    <ol>
      <li>
        <span
          className="ant-breadcrumb-link"
        >
          <a
            href="https://www.google.fr"
          >
            Campaigns
          </a>
        </span>
        <span
          className="ant-breadcrumb-separator"
        >
          <span
            className="mcs-icon "
          >
            <i
              className="mcs-chevron-right"
              style={Object {}}
            />
          </span>
        </span>
      </li>
      <li>
        <span
          className="ant-breadcrumb-link"
        >
          <a
            href="https://www.github.fr"
          >
            Display
          </a>
        </span>
        <span
          className="ant-breadcrumb-separator"
        >
          <span
            className="mcs-icon "
          >
            <i
              className="mcs-chevron-right"
              style={Object {}}
            />
          </span>
        </span>
      </li>
    </ol>
  </nav>
  <div
    className="left-part-margin"
  />
</div>
`;
